tox.ini

```diff
@@ -74,7 +74,17 @@ deps =
 description = Run unit tests
 
 [testenv:integration]
-commands = pytest -v --tb native --asyncio-mode=auto {[vars]tst_path}integration --log-cli-level=INFO -s {posargs}
+commands = 
+    pytest -v --tb native --asyncio-mode=auto {[vars]tst_path}integration/test_deploy_runners.py --keep-models --log-cli-level=INFO -s {posargs}
+    pytest -v --tb native --asyncio-mode=auto {[vars]tst_path}integration/test_charm.py --log-cli-level=INFO -s {posargs}
+deps =
+    -r requirements-integration.txt
+description = Run integration tests
+
+[testenv:aws-integration]
+commands =
+    pytest -v --tb native --asyncio-mode=auto {[vars]tst_path}integration/test_deploy_aws.py --keep-models --log-cli-level=INFO -s {posargs}
+    pytest -v --tb native --asyncio-mode=auto {[vars]tst_path}integration/test_charm.py --log-cli-level=INFO -s {posargs}
```

Comment:
In general, it is really helpful for maintenance if our `tox.ini` doesn't need special environments.  Ideally, we have a common `tox.ini` for all charms so we don't need to maintain multiple (ex: what obs does.  We've tried to do that too but haven't been rigid about it).

This feels like something that could be handled by an argument in pytest passed in through the `{posargs}`

Reply:
This could even still support multiple test files.  The arguments we pass here could be `-k` or `mark`'s.  

A vote in favor of editing the `tox.ini` like you have here though is that it makes it clearer to a user what is available.  Forcing users to pass `-k` and `mark`s makes them *know* they're available - that's not good either

Reply:
Yeah @ca-scribner  we had this discussion in bundle tests release tests where we decided marks are not a good way so we should rather use separate files .... I think separate files are good way here as there might be more testing connected with aws specifically.

Reply:
I think separate environments for AWS and regular integration are cleaner. Also, will allows do changes in tox.ini later on when we add some other test files or settings specific for AWS. Eg. run integration test on different K8S in AWS - CK8S and EKS - then that `integration-aws` environment will take care of the setup for each and will re-run the test.

Reply:
The benefit of consistent tox environments is that the github workflows for integration.yaml are then also common between different charms, meaning we don't need to maintain that file but instead share it across charms (and teams).  This was something @sed-i and I were discussing last week.  For the goal of reducing duplication between our charms and then also other team's charms, having the same tox environments is really helpful

That being said, this is a special case so maybe it makes sense to deviate here.

Reply:
The benefit of consistent tox environments is that we can have common github workflow files (integration.yaml) across our charms (and across teams), reducing our duplication and maintenance.  This was something @sed-i raised recently when we were discussing how to find commonality in our tests.  For the goal of reducing duplication between our charms and then also other team's charms, having the same tox environments is really helpful

That being said, this is a special case so maybe it makes sense to deviate here.

Reply:
Currently, we do not share `integration.yaml` between charms. They all different. There is no plan to unify it at this point. This will be a big effort and should be addressed as a separate roadmap item, speced out, and executed. At this present iteration I think we should proceed without trying to design for the future.